File: backend/src/main/java/com/example/autoproject/entity/Menu.java
package com.example.autoproject.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("sys_menu")
public class Menu {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long parentId;

    private String menuName;

    private Integer menuType;

    private String path;

    private String component;

    private String icon;

    private String permission;

    private Integer sortOrder;

    private Integer status;

    private Integer visible;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;

    @TableField(exist = false)
    private List<Menu> children;
}

File: frontend/src/api/menu.ts
import request from './index'

export interface MenuVO {
  id: number
  parentId: number
  menuName: string
  menuType: number
  path: string
  component: string
  icon: string
  permission: string
  sortOrder: number
  status: number
  visible: number
  createTime: string
  updateTime: string
  children?: MenuVO[]
}

export interface MenuCreateDTO {
  parentId?: number
  menuName: string
  menuType: number
  path?: string
  component?: string
  icon?: string
  permission?: string
  sortOrder?: number
}

export interface MenuUpdateDTO {
  parentId?: number
  menuName: string
  menuType: number
  path?: string
  component?: string
  icon?: string
  permission?: string
  sortOrder?: number
}

export function getMenus() {
  return request.get<MenuVO[]>('/menus')
}

export function createMenu(data: MenuCreateDTO) {
  return request.post<MenuVO>('/menus', data)
}

export function updateMenu(id: number, data: MenuUpdateDTO) {
  return request.put<MenuVO>(`/menus/${id}`, data)
}

export function deleteMenu(id: number) {
  return request.delete(`/menus/${id}`)
}
